testing/functional/tests/cypress/integration/student/02-login-page.spec.ts

```diff
@@ -0,0 +1,220 @@
+
+  beforeEach(() => {
+    cy.visit("/");
+  });
+
+  it("Verify that user able to login with a valid username and valid password.", () => {
+    welcomeObject.virtualTestingButton().should("be.visible").click();
+    welcomeObject.virtualTestingButtonText().should("be.visible");
+    welcomeObject.virtualDeviceId().click({ force: true });
+    cy.wait(2000);
+    loginObject
+      .cardSerialNumberInputText()
+      .type(username)
+      .should("have.value", username);
+    cy.wait(2000);
+    loginObject.cardSerialNumberContinueButton().click();
+    loginObject
+      .passcodeInputText()
+      .type(password)
+      .should("have.value", password);
+    loginObject.passcodeContinueButton().click();
+    loginObject.verifyLoggedInText();
+  });
+
+  it("Verify that user is not able to redirect to next page with invalid Card Serial Number in Virtual Card Testing.", () => {
+    welcomeObject.virtualTestingButton().should("be.visible").click();
+    welcomeObject.virtualTestingButtonText().should("be.visible");
+    welcomeObject.virtualDeviceId().click({ force: true });
+    cy.wait(2000);
+    loginObject.cardSerialNumberInputText().type(invalidUsername);
+    cy.wait(2000);
+    loginObject.cardSerialNumberContinueButton().click();
+    loginObject.cardNotFoundText();
+  });
+
+  it("Test with empty Card Serial Number such that it must get failed.", () => {
+    welcomeObject.virtualTestingButton().should("be.visible").click();
+    welcomeObject.virtualTestingButtonText().should("be.visible");
+    welcomeObject.virtualDeviceId().click({ force: true });
+    cy.wait(2000);
+    loginObject.cardSerialNumberInputText();
+    cy.wait(2000);
+    loginObject.cardSerialNumberContinueButton().click();
+    loginObject.emptyCardNumberText();
+  });
+
+  it("Verify that user can redirect to next page by pressing Enter key after typing in valid Card Serial Number.", () => {
+    welcomeObject.virtualTestingButton().should("be.visible").click();
+    welcomeObject.virtualTestingButtonText().should("be.visible");
+    welcomeObject.virtualDeviceId().click({ force: true });
+    cy.wait(2000);
+    loginObject
+      .cardSerialNumberInputText()
+      .type(username)
+      .should("have.value", username);
+    cy.wait(2000);
+    loginObject
+      .enterKeyFromCardSerialNumber()
+      .should("be.visible")
+      .type("{enter}");
+    loginObject.passcodePageText();
+  });
+
+  it("Verify that Card Serial Number with case sensitivity.", () => {
+    welcomeObject.virtualTestingButton().should("be.visible").click();
+    welcomeObject.virtualTestingButtonText().should("be.visible");
+    welcomeObject.virtualDeviceId().click({ force: true });
+    cy.wait(2000);
+    loginObject.cardSerialNumberInputText().type(caseUsername);
+    cy.wait(2000);
+    loginObject.caseSenstivityCardSerialNumber().should("be.visible");
+  });
+
+  it("Verify that Card Serial Number must be within 15 digits allowed", () => {
```

Comment:
Are we still testing the BCSC page requirements?

Reply:
We have removed all login cases code & only added one valid login case for logged in users.